tests/luagit2_commit_test.lua

```diff
@@ -0,0 +1,234 @@
+-- @module luagit2_commit_test
+local luagit2_commit_tests = {}
+
+function luagit2_commit_tests.test_luagit2_commit_author()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	author_signature = lib.luagit2_commit_author(commit)
+	lunatest.assert_not_nil(author_signature)
+	lunatest.assert_equal("Satyendra_Kumar_Banjare" , lib.luagit2_get_signature_details(author_signature,1))
+	lunatest.assert_equal("satyendrabanjare99@gmail.com" , lib.luagit2_get_signature_details(author_signature,2))
+end
+
+function luagit2_commit_tests.test_luagit2_commit_body()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id_nil_body = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
```

Comment:
Many of the test information is used in different tests and there's a lot of repetition. 
For example, you could open the repo only once before the tests.
 
Additionally, you need to call `luagit2_repository_free` after you finish using the repo in the test or you might get some memory leaks. 

The easiest way to do this, is by using some test helpers like `setup`, `beforeAll`, `afterAll`, `tearDown`, etc. The names can vary between different test libraries. I tried looking up for the api of lunatest and I couldn't find. Could you send it to me? If it doesn't provide these options, then I would recommend switching to a different test library that offers that, like busted  (http://olivinelabs.com/busted/)